provider/github/watcher_test.go

```diff
@@ -121,7 +133,7 @@ func (s *WatcherTestSuite) TestWatch_WithError() {
 		fmt.Fprint(w, `[{"id":1}]`)
 	})
 	s.mux.HandleFunc("/repos/mock/test/events", func(w http.ResponseWriter, r *http.Request) {
-		fmt.Fprint(w, `[{"id":"1", "type":"PushEvent", "payload":{"push_id": 1}}]`)
+		fmt.Fprint(w, `[{"id":1}]`)
```

Comment:
why did you change payload? Now the code is just skipping event, no?

Reply:
With or without the payload the resulting event.ID is the same. Since what we are testing here is what happens when the watch callback returns an error, I though I'd simplify this part that is not adding much to the test.

The event is not skipped, that is done by the callback `(s *Server) handleEvent`, replaced here with our callback that just checks the ID.

Reply:
Sorry. I don't understand how this test works after this change:

- We call [Watch](https://github.com/src-d/lookout/blob/master/provider/github/watcher.go#L86) with custom callback.
- It calls [watchEvents](https://github.com/src-d/lookout/blob/master/provider/github/watcher.go#L131)
- On event (changed above) it calls [handleEvents](https://github.com/src-d/lookout/blob/master/provider/github/watcher.go#L182) which just [castEvent](https://github.com/src-d/lookout/blob/master/provider/github/utils.go#L25)
- without type it return `nil, nil`
- then we go back to [handleEvents](https://github.com/src-d/lookout/blob/master/provider/github/watcher.go#L188) that does `continue` if event is `nil` and never calls callback.

Reply:
Ah. I see how it works now. Test returns review above. Then maybe it’s better to replace this response by empty array to make it clear that callback is never called for events endpoint?

Reply:
Yes, it's working because the path for the pull event enters the callback and returns the foo error. But it's strange to force a path this way, so I think I'll put back the payload and maybe later we can duplicate all tests and run them  with pull or push events.